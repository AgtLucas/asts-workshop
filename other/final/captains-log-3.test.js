import stripIndent from 'strip-indent'
import * as babel from 'babel-core'
import captainsLog from './captains-log-3'

test('transpiles console.log calls to include contextual info', () => {
  const source = stripIndent(
    `
      function add(a, b) {
        console.log(a, b)
        return a + b
      }

      function subtract(a, b) {
        console.log(a, b)
        return a - b
      }

      const multiply = (a, b) => {
        console.log(a, b)
        return a * b
      }
      add(1, 2)
      subtract(2, 1)
      multiply(3, 4)
      console.log('sup dawg')
    `,
  ).trim()
  const code = transpile(source)
  expect(code).toMatchSnapshot()
})

function transpile(source) {
  const {code} = babel.transform(source, {
    babelrc: false,
    plugins: [captainsLog],
  })
  return code
}
