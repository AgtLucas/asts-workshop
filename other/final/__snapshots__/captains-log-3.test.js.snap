// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transpiles console.log calls to include contextual info 1`] = `
"function add(a, b) {
  console.log('add 2:2', a, b);
  return a + b;
}

function subtract(a, b) {
  console.log('subtract 7:2', a, b);
  return a - b;
}

const multiply = (a, b) => {
  console.log('multiply 12:2', a, b);
  return a * b;
};
add(1, 2);
subtract(2, 1);
multiply(3, 4);
console.log('18:0', 'sup dawg');"
`;
